### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by LeonIllya**

### CODE REVIEW

File: src/main/java/book/store/config/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    private final UserDetailsService userDetailsService;
+
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+        return http
+            .cors(AbstractHttpConfigurer::disable)
+            .csrf(AbstractHttpConfigurer::disable)
+            .authorizeHttpRequests(
+                auth -> auth.requestMatchers("/auth/**", "/error", "/swagger-ui/**")
+                .permitAll().anyRequest().authenticated()
```
Comment (olekskov) [inline src/main/java/book/store/config/SecurityConfig.java:29]: Please try to format code so it has new line between logical compositions, something like
```suggestion
            .authorizeHttpRequests(auth -> auth
                .requestMatchers("/auth/**", "/error", "/swagger-ui/**").permitAll()
                .anyRequest().authenticated()
```

File: src/main/java/book/store/controller/BookController.java [hunk 2]
```diff
@@ -18,40 +22,53 @@
...
 import org.springframework.web.bind.annotation.RestController;
 
+@Tag(name = "Book management", description = "Endpoint for managing books")
 @RequiredArgsConstructor
 @RestController
 @RequestMapping(value = "books")
 public class BookController {
     private final BookService bookService;
 
+    @PreAuthorize("hasRole('ROLE_USER')")
```
Comment (olekskov) [inline src/main/java/book/store/controller/BookController.java:32]: Please doublecheck it works.
As far as I remember hasRole command automatically should prepend value with ROLE_ prefix, so I would expect more
```suggestion
    @PreAuthorize("hasRole('USER')")
```

File: src/main/java/book/store/controller/BookController.java [hunk 3]
```diff
@@ -18,40 +22,53 @@
...
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.getAll();
+    @Operation(summary = "Get all books", description = "Get a list of all books")
+    public List<BookDto> getAll(Pageable pageable) {
+        return bookService.getAll(pageable);
     }
 
+    @PreAuthorize("hasRole('ROLE_USER')")
```
Comment (olekskov) [inline src/main/java/book/store/controller/BookController.java:39]: please make sure you really wish to close this endpoint from ADMIN user
If you wish both users to be able to reach it then just remove annotation

**Review by olekskov — CHANGES_REQUESTED** [review]

File: src/main/java/book/store/dto/user/UserResponseDto.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import lombok.Data;
+
+@Data
+public class UserResponseDto {
+    private Long id;
+    private String email;
+    private String firstName;
+    private String lastName;
+    private String shippingAddress;
+
```
Comment (okuzan) [inline src/main/java/book/store/dto/user/UserResponseDto.java:12]:
```suggestion
```

File: src/main/java/book/store/model/Role.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import jakarta.persistence.Entity;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import lombok.Data;
+import org.springframework.security.core.GrantedAuthority;
+
+@Data
```
Comment (okuzan) [inline src/main/java/book/store/model/Role.java:13]: avoid using @ Data on JPA entities, it has side effects (toString, equalsAndHashcode specifically)

File: src/main/java/book/store/model/User.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+@Entity(name = "users")
+@Data
+@SQLDelete(sql = "UPDATE users SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
```
Comment (okuzan) [inline src/main/java/book/store/model/User.java:23]: Where is deprecated, we should use SQLRestriction

File: src/main/java/book/store/repository/role/RoleRepository.java [hunk 7]
```diff
@@ -0,0 +1,9 @@
+package book.store.repository.role;
+
+import book.store.model.Role;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface RoleRepository extends JpaRepository<Role, Long> {
+
```
Comment (okuzan) [inline src/main/java/book/store/repository/role/RoleRepository.java:7]:
```suggestion
```

File: src/main/resources/db/changelog/changes/05-insert-users-in-table.yaml [hunk 8]
```diff
@@ -0,0 +1,21 @@
+databaseChangeLog:
+  - changeSet:
+      id: insert-users-in-table
+      author: LeonIlya
+      changes:
+        - insert:
+            tableName: users
+            columns:
+              - column: { name: email, value: "admin1@gmail.com" }
+              - column: { name: password, value: "$2a$10$zwE7/e4b0WxpHlkCHqzoiO0gVJOn0AtQWBHbdFUe6Dqk84Aq1/j26" }
```
Comment (okuzan) [inline src/main/resources/db/changelog/changes/05-insert-users-in-table.yaml:10]: include password in plaintext as comment, rn it's unusable for testing

**Review by okuzan — CHANGES_REQUESTED** [review]
Good job overall!

File: src/main/java/book/store/model/User.java [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+import org.hibernate.annotations.Where;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+@Entity(name = "users")
+@Data
```
Comment (kerrrusha) [inline src/main/java/book/store/model/User.java:22]: dont use data with entity as it cause bugs
```suggestion
@Getter
@Setter
```

**Review by kerrrusha — APPROVED** [review]

**Pull request merged**